lms/djangoapps/shoppingcart/models.py

```diff
@@ -1196,7 +1197,7 @@ def is_registration_code_redeemed(cls, course_reg_code):
         Checks the existence of the registration code
         in the RegistrationCodeRedemption
         """
-        return cls.objects.filter(registration_code=course_reg_code).exists()
+        return cls.objects.filter(registration_code__code=course_reg_code).exists()
```

Comment:
@chrisndodge nope this wasn't a bug, We were passing the registration_code object before. In the current story we just have only a registration_code, so i make adjustments in the classmethod to lookup for registration_code.